### PR TITLE
feat: change layout of date picker, granularity and data zoom

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: [#12717](https://github.com/lightdash/lightdash/issues/12717)

### Description:

- date filter is a segmented control
- presets reference a main util to get the default date range for an interval/granularity
- all controls are above the viz

https://github.com/user-attachments/assets/430e98fc-b799-4489-af61-8daa1647de67

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts [hunk 1]
```diff
@@ -106,5 +106,6 @@ export const useRunMetricExplorerQuery = ({
             !!metricName &&
             !!comparison &&
             !!dateRange,
+        keepPreviousData: true,
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts:109]: smoother experience to the user

Comment (joaoviana) [reply packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts:109]: we just have to be careful with isLoading usage and utilise isFetching when we do better loading states

File: packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx [hunk 2]
```diff
@@ -157,91 +198,102 @@ const MetricsVisualization: FC<Props> = ({ data, comparison }) => {
...
-                    onMouseMove={handleMouseMove}
-                    onMouseUp={handleMouseUp}
-                >
-                    <CartesianGrid
-                        horizontal
-                        vertical={false}
-                        stroke={colors.gray[2]}
-                        strokeDasharray="4 3"
-                    />
+            {showEmptyState && <MetricsVisualizationEmptyState />}
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx:201]: empty state is now in metrics visualization, as well as the date filters' components.

File: packages/frontend/src/features/metricsCatalog/components/MetricPeekDatePicker.tsx [hunk 3]
```diff
@@ -56,57 +57,123 @@ export const MetricPeekDatePicker: FC<Props> = ({
...
-                        radius="md"
-                        onClick={() => handleOpen(!isOpen)}
-                        size="sm"
-                        px="xs"
-                        w="100%"
+                <Group position="apart" w="fill-available">
+                    <SegmentedControl
+                        size="xs"
+                        h={32}
+                        data={customWithPresets}
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/MetricPeekDatePicker.tsx:105]: moving this to a separate component on a separate PR - soon

File: packages/frontend/src/features/metricsCatalog/utils/metricPeekDate.tsx [hunk 4]
```diff
@@ -195,32 +198,84 @@ export const getDateRangePresets = (
...
+        case TimeFrames.YEAR:
+            isSameTimeInterval = 'year';
+            break;
+        default:
+            isSameTimeInterval = undefined;
+    }
+
+    return (
+        dayjs(range1[0]).isSame(dayjs(range2[0]), isSameTimeInterval) &&
+        dayjs(range1[1]).isSame(dayjs(range2[1]), isSameTimeInterval)
```
Comment (IrakliJani) [inline packages/frontend/src/features/metricsCatalog/utils/metricPeekDate.tsx:265]: oh nice, I did not know this (interval bit)

**Review by IrakliJani — APPROVED** [review]
